workspace/src/common/launch/art_localization_launch/launch/art_localization.launch.py

```diff
@@ -48,9 +48,11 @@ def generate_launch_description():
     AddLaunchArgument(ld, "art_localization/input/magnetometer", "/sensing/magnetometer/data")
     AddLaunchArgument(ld, "art_localization/input/gyroscope", "/sensing/gyroscope/data")
     AddLaunchArgument(ld, "art_localization/input/accelerometer", "/sensing/accelerometer/data")
-    AddLaunchArgument(ld, "art_localization/output/vehicle_state", "/vehicle/filtered_state")
+    AddLaunchArgument(ld, "art_localization/input/vehicle_inputs", "/control/vehicle_inputs")
+    AddLaunchArgument(ld, "art_localization/input/ground_truth", "/vehicle/state")
+    AddLaunchArgument(ld, "art_localization/output/filtered_state", "/vehicle/filtered_state")
 
-    AddLaunchArgument(ld, "vis", "False")
+    AddLaunchArgument(ld, "estimation_alg", "ground_truth", choices=("ground_truth", "extended_kalman_filter", "particle_filter"))
```

Comment:
nit: not needed for me to approve, but I think a good design choice would be to have an enum in the class that uses these params (in this case state_estimation) that has all the eligible choices. Then here you just dynamically load all the enum options.

This way you can just update the enum and don't have to update multiple files. 

Again, not needed for this PR, just a thought in the future.

Reply:
I'm not exactly sure what this would look like... could you point me towards an example?

Reply:
Like this:

```python
# state_estimation.py

from enum import Enum
class EstimationAlgorithmOption(Enum):
    GROUND_TRUTH = "ground_truth"
    EXTENDED_KALMAN_FILTER = "extended_kalman_filter"
    PARTICLE_FILTER = "particle_filter"

# ... later ...

if self.estimation_alg == EstimationAlgorithmOption.GROUND_TRUTH:
  # ... stuff ...
elif self.estimation_alg == EstimationAlgorithmOption.EXTENDED_KALMAN_FILTER:
  # ... stuff ...
...
```

```python
# art_localization.launch.py

from localization_py.state_estimation import EstimationAlgorithmOption

# ...

    AddLaunchArgument(ld, "estimation_alg", EstimationAlgorithmOption.GROUND_TRUTH, choices=[o.value for o in EstimationAlgorithmOption])
```

Just my opinion. Helps down the line when/if you add other algos. Up to you whether you want something like this or not.

Reply:
Making this change rn. I can see how this is nice for not needing to change the launch file in the future, but that seems like a very small amount of work to me, especially compared to the if statements that depend on the options actually existing (In `if self.estimation_alg == EstimationAlgorithmOption.GROUND_TRUTH` the `GROUND_TRUTH` is still a hardcoded variable name). And adding a class seems like it doesn't add a ton of clarity, but does add some overhead towards having an extra class defined. Are there other reasons I am not seeing for doing this in general?

Reply:
also how do I get the estimation algorithm parameter from the launch config when doing this?

Reply:
Well don't add it if you don't think it makes sense. But imo, it's better to read `if self.estimation_alg == EstimationAlgorithmOption.GROUND_TRUTH` than `if self.estimation_alg == "ground_truth"` cause without an enum, you don't actually know what all the choices ever are. imo centralizing the definition/options associated with some logic is the main reason for enum, and so having the extra class is basically a documentation element that you can refer to to know what is available. If an enum option doesn't exist, your type checker/vscode should tell you that (and this is different from if they were just strings cause that's not possible in this case, but it still could not be correct as an option). 

Thoughts?

Reply:
I also get the following error when using your code... 
`[ERROR] [launch]: Caught exception in launch (see debug for traceback): 'EstimationAlgorithmOption' object is not iterable`. This is from the launchfile.

Reply:
<img width="495" alt="Screenshot 2023-10-09 at 16 20 55" src="https://github.com/uwsbel/autonomy-research-testbed/assets/43622931/d520b0f7-c764-472e-900d-46868f31411a">

idk if this is what the error is related to, but I can iterate over the enum class.